hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/impl/CryptoPreTransactionHandlerImpl.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.token.impl;
+
+import static com.hedera.services.legacy.proto.utils.CommonUtils.extractTransactionBody;
+import static com.hedera.services.utils.MiscUtils.asUsableFcKey;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.hedera.node.app.service.token.CryptoPreTransactionHandler;
+import com.hedera.node.app.spi.meta.TransactionMetadata;
+import com.hedera.node.app.spi.meta.impl.InvalidTransactionMetadata;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.Transaction;
+
+/**
+ * An implementation of {@code CryptoPreTransactionHandler} for validating all transactions defined
+ * in the protobuf "CryptoService" in pre-handle. It adds all the verified signatures to the
+ * including signature verification.
+ */
+public class CryptoPreTransactionHandlerImpl implements CryptoPreTransactionHandler {
+    private final AccountStore accountStore;
+
+    public CryptoPreTransactionHandlerImpl(final AccountStore accountStore) {
+        this.accountStore = accountStore;
```

Comment:
Let's use Objects.requireNonNull on the passed in accountStore to make sure it is not null, and also use the @Nonnull annotation

Reply:
Added . Thanks